homeassistant/components/ps4/config_flow.py

```diff
@@ -160,7 +160,7 @@ def __init__(self):
             list(self.device_list))
         link_schema[vol.Required(CONF_REGION)] = vol.In(list(regions))
         link_schema[vol.Required(CONF_CODE)] = vol.All(
-            vol.Strip, vol.Length(min=8, max=8), vol.Coerce(int)
+            vol.Strip, vol.Length(min=8, max=8), vol.Coerce(str)
```

Comment:
Shouldn't we first coerce to integer and then coerce to string? Otherwise we won't make sure that the code is a number.

Reply:
I tried adding the str coerce it to the end of the vol.All tuple, but it overrides the int coerce, allowing non-int to be entered.

Reply:
Ah, ok, that's a limitation of how voluptuous-serialize has implemented the serialization of voluptuous. It's not how voluptuous works.

Yes, then just copy to string as done before.

Reply:
Ok cool!